packages/common/core-interfaces/src/test/jsonStringifyAndParse.spec.ts

```diff
@@ -0,0 +1,2700 @@
+	objectWithSelfReference,
+	objectWithSymbolOrRecursion,
+	objectWithUnknownAdjacentToOptionalRecursion,
+	// objectWithOptionalUnknownAdjacentToOptionalRecursion,
+	objectWithUnknownInOptionalRecursion,
+	// objectWithOptionalUnknownInOptionalRecursion,
+	selfRecursiveFunctionWithProperties,
+	selfRecursiveObjectAndFunction,
+	objectInheritingOptionalRecursionAndWithNestedSymbol,
+	simpleJson,
+	simpleImmutableJson,
+	jsonObject,
+	immutableJsonObject,
+	classInstanceWithPrivateData,
+	classInstanceWithPrivateMethod,
+	classInstanceWithPrivateGetter,
+	classInstanceWithPrivateSetter,
+	classInstanceWithPublicData,
+	classInstanceWithPublicMethod,
+	objectWithClassWithPrivateDataInOptionalRecursion,
+	functionObjectWithPrivateData,
+	functionObjectWithPublicData,
+	classInstanceWithPrivateDataAndIsFunction,
+	classInstanceWithPublicDataAndIsFunction,
+	mapOfStringsToNumbers,
+	readonlyMapOfStringsToNumbers,
+	setOfNumbers,
+	readonlySetOfNumbers,
+	brandedNumber,
+	brandedString,
+	brandedObject,
+	brandedObjectWithString,
+	objectWithBrandedNumber,
+	objectWithBrandedString,
+	brandedStringIndexOfBooleans,
+	brandedStringAliasIndexOfBooleans,
+	brandedStringRecordOfBooleans,
+	brandedStringAliasRecordOfBooleans,
+	brandedStringIndexOfNumbers,
+	brandedStringAliasIndexOfNumbers,
+	brandedStringRecordOfNumbers,
+	brandedStringAliasRecordOfNumbers,
+	brandedStringAliasIndexOfTrueOrUndefined,
+	datastore,
+	fluidHandleToNumber,
+	objectWithFluidHandle,
+	// objectWithFluidHandleOrRecursion,
+	opaqueSerializableObject,
+	opaqueDeserializedObject,
+	opaqueSerializableAndDeserializedObject,
+	opaqueSerializableUnknown,
+	opaqueDeserializedUnknown,
+	opaqueSerializableAndDeserializedUnknown,
+	objectWithOpaqueSerializableUnknown,
+	objectWithOpaqueDeserializedUnknown,
+	opaqueSerializableInRecursiveStructure,
+	opaqueDeserializedInRecursiveStructure,
+	opaqueSerializableAndDeserializedInRecursiveStructure,
+	opaqueSerializableObjectRequiringBigintSupport,
+	opaqueDeserializedObjectRequiringBigintSupport,
+	opaqueSerializableAndDeserializedObjectRequiringBigintSupport,
+	opaqueSerializableObjectExpectingBigintSupport,
+	opaqueDeserializedObjectExpectingBigintSupport,
+	opaqueSerializableAndDeserializedObjectExpectingBigintSupport,
+	jsonStringOfString,
+	jsonStringOfObjectWithArrayOfNumbers,
+	jsonStringOfStringRecordOfNumbers,
+	jsonStringOfStringRecordOfNumberOrUndefined,
+	jsonStringOfBigInt,
+	jsonStringOfUnknown,
+} from "./testValues.js";
+
+import type {
+	JsonString,
+	JsonStringifyOptions,
+} from "@fluidframework/core-interfaces/internal";
+import { JsonParse, JsonStringify } from "@fluidframework/core-interfaces/internal";
+import type {
+	JsonDeserialized,
+	JsonSerializable,
+	JsonTypeWith,
+	NonNullJsonObjectWith,
+	OpaqueJsonDeserialized,
+} from "@fluidframework/core-interfaces/internal/exposedUtilityTypes";
+
+/**
+ * !!! TODO update: Defined using `JsonSerializable` type filter tests `JsonStringify` at call site.
```

Comment:
I'm not sure what this sentence means, but maybe this todo was supposed to be address in this pr? If not, adjust the phrasing for clarity.